aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -617,6 +647,195 @@ module aptos_framework::coin {
         let BurnCapability<CoinType> {} = burn_cap;
     }
 
+    /// Initialize fungible asset migration.
+    public entry fun initialize_fungible_asset_migration(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        move_to(aptos_framework, FungibleAssetMigration {
+            coins_to_fungible_assets: smart_table::new(),
+            fungible_assets_to_coins: smart_table::new(),
+            enable_default_fungible_assets_in_transfers: false,
+        });
+    }
+
+    /// Create a fungible asset version of an existing Coin. This can only be called by
+    /// the signer corresponding to the address where CoinType is defined/deployed.
+    /// Icon and project uris are new fields in Fungible Asset that don't exist in Coin
+    /// and can be provided or left empty at the caller's discretion.
+    public entry fun initialize_fungible_asset<CoinType>(
+        creator: &signer,
+        maximum_supply: Option<u128>,
+        icon_uri: String,
+        project_uri: String,
+    ) acquires CoinInfo, FungibleAssetMigration {
+        assert!(coin_address<CoinType>() == signer::address_of(creator), EFUNGIBLE_ASSET_MIGRATION_UNAHORIZED);
+
+        // TODO: should use aptos_framework as the signer, but no idea how to get aptos_framework signer
```

Comment:
@movekevin @lightmark instead of `create_named_object(creator, seed)`, I think it makes more sense to do `create_named_object(aptos_framework, seed)`.

however, i have no clue how i can access `aptos_framework: &signer` - any idea?

Reply:
`aptos_framework::create_signer::creator_signer` has to friend this module.